### PR TITLE
rh config support for specifying launcher type

### PR DESCRIPTION
**Pull request by jlewitt1**

### CODE REVIEW

**Comment by jlewitt1** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1388?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1388** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1388?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1336** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1336?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 5 other dependent PRs ([#1343](https://github.com/run-house/runhouse/pull/1343) <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1343?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#1373](https://github.com/run-house/runhouse/pull/1373) <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1373?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#1374](https://github.com/run-house/runhouse/pull/1374) <a href="https://app.graphite.dev/github/pr/run-house/runhouse/1374?utm_source=stack-comment-icon" target="_blank">Graphite</a> and 2 others)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @jlewitt1 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: runhouse/rns/defaults.py [hunk 1]
```diff
@@ -279,6 +295,16 @@ def delete_defaults(self, config_path: Optional[str] = None):
         except OSError:
             raise Exception(f"Failed to delete config file from path {config_path}")
 
+    def set_default_launcher_type(self, launcher_type):
```
Comment (carolineechen) [inline runhouse/rns/defaults.py:298]: are there individual use cases for the `launcher_type` property setter, and actually setting it in the `default_cache` or could we just combine them and have the setter also update the `default_cache`?

File: runhouse/resources/hardware/on_demand_cluster.py [hunk 2]
```diff
@@ -527,7 +527,7 @@ def up(self, verbose: bool = True, force: bool = False):
             logger.info("Launching cluster with Den")
             DenLauncher.up(cluster=self, verbose=verbose, force=force)
 
-        else:
+        elif self.launcher_type == LauncherType.LOCAL:
```
Comment (carolineechen) [inline runhouse/resources/hardware/on_demand_cluster.py:530]: L109 in this file should be `self.launcher_type = launcher_type or configs.launcher_type`

File: runhouse/rns/defaults.py (hunk 1 above)
Comment (jlewitt1) [reply runhouse/rns/defaults.py:298]: im not reallty sure we need it at all actually, this doesn't feel like the kind of thing a user would change all that much. i think i'll removce it

Comment (carolineechen) [reply runhouse/rns/defaults.py]: the launcher_type setting only sets the variable and not value in the defaults cache though so then there's no way to set it right? do you think this make more sense to propagate through and save it down?
```
@launcher_type.setter
    def launcher_type(self, value):
        self._launcher_type = value
        self.set("launcher_type", value)
```

Comment (jlewitt1) [reply runhouse/rns/defaults.py]: oh i see, yea this is prob a config type we want to persist, so may as well save it down

**Review by carolineechen — APPROVED** [review]

**Pull request merged**